polkadot/node/network/approval-distribution/src/lib.rs

```diff
@@ -2255,18 +2307,42 @@ impl State {
 			&self.topologies,
 			|block_entry| {
 				let block_age = max_age - block_entry.number;
+				// We want to resend only for blocks between min_age and min_age +
```

Comment:
I believe it should be sufficient that we trigger aggression only for oldest unfinalized block. From what I've seen in production and testing is that usuallt are a few unapproved candidates holding finality on a particular block. 

Resending for all unfinalized blocks doesn't really make sense, it's better to make small incremental progress than send large bursts of messages.  So I would just remove the `resend_unfinalized_period` parameter and code.

Am I missing anything ?

Reply:
> Am I missing anything ?
I don't think you are, I was think about the same thing, but I did not know the initial reason for using `resend_unfinalized_period` so I was trying to avoid it entirely, so yeah I'll look into resending it just for the last unfinalized.